FirstAnalysis/efficiency_studies.py

```diff
@@ -1,22 +1,78 @@
 #!/usr/bin/env python
-from ROOT import TH1F, TCanvas, TEfficiency, TFile, TLegend, gPad, gStyle
 
+# pylint: disable=missing-module-docstring, missing-function-docstring
 
-def saveCanvas(canvas, title):
+import argparse
+
+from ROOT import TH1F, TCanvas, TEfficiency, TFile, TLegend, TLatex # pylint: disable=import-error,no-name-in-module
+from ROOT import gPad, gStyle, gROOT # pylint: disable=import-error,no-name-in-module
+
+def save_canvas(canvas, title):
     format_list = [".png", ".pdf", ".root"]
-    for fileFormat in format_list:
-        canvas.SaveAs(title + fileFormat)
+    for file_format in format_list:
+        canvas.SaveAs(title + file_format)
 
 
-def efficiencytracking(var):
-    # plots the efficiency vs pT, eta and phi for all the species(it extracts the
-    # Efficiency from qa - tracking - efficiency if you have ran with-- make - eff)
-    hadron_list = ["pion", "proton", "kaon", "electron", "muon"]
-    color_list = [1, 2, 4, 6, 8]
-    marker_list = [20, 21, 22, 34, 45]
-    fileo2 = TFile("../codeHF/AnalysisResults_O2.root")
+def prepare_canvas(var, sign, had, det): # pylint: disable=too-many-locals
+    cname = f"c_{had}_{det}_{sign}_{var}"
+    hname = f"hempty_{had}_{det}_{sign}_{var}"
+    ctitle = f"Efficiency for {sign} {had}, {det}"
+    if var == "Pt":
+        ctitle = f"{ctitle} primaries"
+
+    def get_pt_hist():
+        hempty = TH1F(hname, f"{ctitle};Transverse Momentum (GeV/c);Efficiency", 100, 0.05, 10)
+        gPad.SetLogx()
+        return hempty
+    def get_eta_hist():
+        return TH1F(hname, f"{ctitle};Pseudorapidity;Efficiency", 100, -1.5, 1.5)
+    def get_phi_hist():
+        return TH1F(hname, f"{ctitle};Azimuthal angle (rad);Efficiency", 100, 0.0, 6.0)
+
+    hists = {"Pt": get_pt_hist, "Eta": get_eta_hist, "Phi": get_phi_hist}
```

Comment:
Why do you create all histograms if you use only one?

Reply:
I create a selected histogram only at the moment I select the function in line 40. Only then the function is actually called.
Previously, this macro had if-elses depending on `var` to choose a correct histogram.